### DISCUSSION TITLE
GroupBox and Elevation

### DISCUSSION THREAD

**Discussion by Frillaut** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3201] [Q&A] [answered]
Hi!
In this code, `Grid.Background` overlaps `ElevationAssist.Elevation`...

```xaml
<GroupBox Style="{StaticResource MaterialDesignGroupBox}"
          Padding="0" BorderThickness="0"
          materialDesign:ElevationAssist.Elevation="Dp24"
          Header="GroupBox"
          >
    <!--<Grid Background="Transparent"/>-->
    <Grid Background="White"/>
</GroupBox>
```

I tried to play with `Panel.ZIndex`, but it didn't work out for me. Any ideas?

**Comment by nicolaihenriksen** [comment]
@Frillaut Just ha a quick look. I think the support for `ElevationAssist.Elevation` is wrongfully implemented on the `GroupBox` styles. It seems it only applies to the "header". In the `MaterialDesignCardGroupBox` it seems strange to apply it to the header (see image). I think the logical solution would be to pass the elevation down to the nested `Card` One or more issues regarding this could/should be raised so we can get them fixed. I don't have time today, but I may work on it when I have some time.

![image](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/19572699/cedaa73f-6c36-4b72-bc9e-460e0bc92252)

**Comment by nicolaihenriksen** [comment]
@Frillaut Apparently the "shadow only on the header" is a conscious choice. It is even demonstrated in the demo app.

I may still work on it to give the user a bit more flexibility.

![image](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/19572699/617ad64b-4d55-4934-b94d-ba0e6e8b92ee)

**Reply by nicolaihenriksen** [reply, marked as answer]
@Frillaut I submitted a PR https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/pull/3217 which improves on this a bit. Could you take a look to see if the elevation/shadow in the PR seems to align with your expectations? Thanks.

**Reply by Frillaut** [reply]
Yes. The element has become more flexible. Thanks too for your work.
